network_simulator/statistics.py

```diff
@@ -4,6 +4,186 @@
 """
 
 
+from .host import *
+from .link import *
+from .flow import *
+
+
+class LinkStats(object):
+    """
+    Contains information of a link regarding to per-link buffer occupancy,
+    packet loss, and flow rate.
+
+    :ivar list buffer_occupancy: a list of (timestamp, buffer_occupancy)
+    tuples.
+    :ivar list packet_loss_times: a list of timestamps, each corresponding
+    to a packet loss.
+    :ivar list packet_transmit_times: a list of timestamps corresponding
+    to when a packet was transmitted.
+    """
+    def __init__(self, buffer_occupancy=0, packet_loss_times=[],
+                 packet_transmit_times=[]):
+        self.buffer_occupancy = buffer_occupancy
+        self.packet_loss_times = packet_loss_times
+        self.packet_transmit_times = packet_transmit_times
+
+    # TODO(sharon): functions and subclasses/events (if any) goes here.
+
+
+class FlowStats(object):
+    """
+    Contains information of a flow regarding to per-flow send/receive rate
+    and packet round-trip delay.
+
+    :ivar list packet_sent_times_sec: a list of timestamps corresponding to
+    when a data packet was sent.
+    :ivar list packet_rec_times_sec: a list of timestamps corresponding to
+    when an ACK packet was received for a given :class:`.Flow`.
+    :ivar list packet_rtts_sec: a list of RTTs (in seconds).
+    """
+    def __init__(self, packet_sent_times_sec=[], packet_rec_times_sec=[],
+                 packet_rtts_sec=[]):
+        packet_sent_times_sec = packet_sent_times_sec
+        packet_rec_times_sec = packet_rec_times_sec
+        packet_rtts_sec = packet_rtts_sec
+
+    # TODO(sharon): functions and subclasses/events (if any) goes here.
+
+
+class HostStats(object):
+    """
+     Contains information of a host regarding to per-host send/receive rate.
+
+    :ivar list packet_sent_times_sec: a list of (timestamp, packet_size)
+    tuples.
+    :ivar list packet_rec_times_sec: a list of (timestamp, packet_size)
+    tuples.
+    """
+    def __init__(self, packet_sent_times_sec=[], packet_rec_times_sec=[]):
```

Comment:
Might be a good idea to remove the _sec from the variable names, and just specify that the timestamps are in seconds and the packet sizes are in bits in the :ivar documentation above.